lib/cartopy/crs.py

```diff
@@ -529,10 +529,21 @@ def filter_last(t):
                                            're-added. Please raise an issue.')
 
         # filter out any non-valid linear rings
+        def ring_validity(linear_ring):
```

Comment:
The input here is a `LineString` before converting to `LinearRing`, so I find this naming confusing (and yes, the original list comprehension was confusingly named too.)

Reply:
I've changed the name to `line_string`, which ought to better reflect what's going on.

Reply:
I think the comment also about the name of the function being `ring_validity` when it's working with a line string.